imagetest/test_suites/network/dhcp_test.go

```diff
@@ -13,40 +15,151 @@ import (
+	}
+
+	if err != nil {
+		t.Fatalf("dhcp command failed or not found: %v", err)
+	}
+}
+
+func parseNetworkctlOutput(cmd *exec.Cmd) error {
+	out, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("networkctl command failed %v", err)
+	}
+
+	// check for any line with dhcpv4. If the line is found, check that
+	// a valid IP address is found in the same line.
+	for _, line := range strings.Split(string(out), "\n") {
+		upperLine := strings.ToUpper(line)
+		if strings.Contains(upperLine, "DHCPV4") {
+			for _, token := range strings.Fields(upperLine) {
+				if validIPOrCIDR(token) {
+					return nil
+				}
+			}
+		}
+	}
+
+	return fmt.Errorf("valid ip4 address not found in networkctl output")
+}
+
+func parseNmcliOutput(cmd *exec.Cmd) error {
+	out, err := cmd.Output()
 	if err != nil {
-		t.Fatalf("couldn't get image from metadata")
+		return fmt.Errorf("nmcli command failed %v", err)
+	}
+
+	// check for any line with dhcpv4. If the line is found, check that
+	// a valid IP address is found in the same line.
+	for _, line := range strings.Split(string(out), "\n") {
+		upperLine := strings.ToUpper(line)
+		if strings.Contains(upperLine, "IP4.ADDRESS") {
+			for _, token := range strings.Fields(upperLine) {
+				if validIPOrCIDR(token) {
+					return nil
+				}
+			}
+		}
 	}
 
-	// Which dhcp client the guest agent uses for multinic bringup.
-	var dhcpclient string
-	switch {
-	case strings.Contains(image, "sles"):
-		dhcpclient = wicked
-	default:
-		dhcpclient = dhclient
+	return fmt.Errorf("IPV4Address not found in nmcli output")
+}
+
+func parseWickedOutput(cmd *exec.Cmd) error {
+	out, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("wicked command failed %v", err)
 	}
 
+	for _, line := range strings.Split(string(out), "\n") {
+		upperLine := strings.ToUpper(line)
+		if strings.Contains(upperLine, "IPV4") && strings.Contains(upperLine, "DHCP") {
+			for _, token := range strings.Fields(upperLine) {
+				if validIPOrCIDR(token) {
+					return nil
+				}
+			}
+		}
+	}
+
+	return fmt.Errorf("dhcpv4 or ip address not found in wicked output")
+}
+
+func checkDHCPProcess() error {
 	iface, err := utils.GetInterface(1)
 	if err != nil {
-		t.Fatalf("couldn't get secondary interface: %v", err)
+		return fmt.Errorf("couldn't get secondary interface: %v", err)
 	}
 
 	cmd := exec.Command("ps", "x")
 	out, err := cmd.Output()
 	if err != nil {
-		t.Fatalf("ps command failed %v", err)
+		return fmt.Errorf("ps command failed %v", err)
 	}
 	for _, line := range strings.Split(string(out), "\n") {
-		if strings.Contains(line, fmt.Sprintf("%s %s", dhcpclient, iface.Name)) {
-			return
+		if strings.Contains(line, fmt.Sprintf("dhclient %s", iface.Name)) {
+			return nil
 		}
 	}
-	t.Fatalf("failed finding dhcp client process")
+	return fmt.Errorf("failed finding dhclient process")
+}
+
+// accepts IP addresses in the form of a.b.c.d or a.b.c.d/IPNET
+func validIPOrCIDR(token string) bool {
+	fmt.Printf("token is %s", token)
```

Comment:
please use `t.Logf()`

Reply:
this print statement was used for debugging but is unnecessary in the test, removed.